### PR TITLE
Add a note about scale to zero in the dev guide

### PR DESCRIPTION
**Pull request by rakelkar**
**What this PR does / why we need it**:
Add a note to explain why there may be no pods. If you dont run get pods immediately it is a little disconcerting to see no output (as it makes you think your install is broken)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/146)

### CODE REVIEW

File: docs/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -158,10 +158,16 @@ kubectl apply -f docs/samples/tensorflow/tensorflow.yaml
...
 ```console
+$ kubectl get kfservice -n default
+NAME             URL                                  DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
+flowers-sample   flowers-sample.default.example.com   100                                1h
+
 $ kubectl get pods -n default -l serving.kubeflow.org/kfservice=flowers-sample
 NAME                                                READY   STATUS    RESTARTS   AGE
 flowers-sample-default-htz8r-deployment-8fd979f9b-w2qbv   3/3     Running   0          10s
 ```
+NOTE: KFService scales pods to 0 in the absence of traffic. If you dont see any pods, try sending out a query via curl using instructions in the tensorflow sample: https://github.com/kubeflow/kfserving/tree/master/docs/samples/tensorflow
```
Comment (ellis-bigelow) [inline docs/DEVELOPER_GUIDE.md]: nit: KFService -> KFServing or KFService -> KFServices

File: docs/DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -158,10 +158,16 @@ kubectl apply -f docs/samples/tensorflow/tensorflow.yaml
...
 ```console
+$ kubectl get kfservices -n default
+NAME             URL                                  DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
+flowers-sample   flowers-sample.default.example.com   100                                1h
+
 $ kubectl get pods -n default -l serving.kubeflow.org/kfservice=flowers-sample
 NAME                                                READY   STATUS    RESTARTS   AGE
 flowers-sample-default-htz8r-deployment-8fd979f9b-w2qbv   3/3     Running   0          10s
 ```
+NOTE: KFServing scales pods to 0 in the absence of traffic. If you dont see any pods, try sending out a query via curl using instructions in the tensorflow sample: https://github.com/kubeflow/kfserving/tree/master/docs/samples/tensorflow
```
Comment (hougangliu) [inline docs/DEVELOPER_GUIDE.md]: s/dont/don't
s/any pods/any pod

**Pull request merged**